src/prefect/agent/kubernetes/agent.py

```diff
@@ -183,6 +188,60 @@ def manage_jobs(self) -> None:
                                         delete_job = True
                                         break
 
+                            # Report recent events for pending pods to flow run logs
+                            if pod.status.phase == "Pending":
+                                if not self.job_pod_event_timestamps.get(job_name):
+                                    self.job_pod_event_timestamps[job_name] = {}
+
+                                if not self.job_pod_event_timestamps[job_name].get(
+                                    pod_name
+                                ):
+                                    self.job_pod_event_timestamps[job_name][
+                                        pod_name
+                                    ] = datetime.min.replace(tzinfo=pytz.UTC)
+
+                                pod_events = self.core_client.list_namespaced_event(
+                                    namespace=self.namespace,
+                                    field_selector="involvedObject.name={}".format(
+                                        pod_name
+                                    ),
+                                    timeout_seconds=30,
+                                )
+
+                                for event in sorted(
+                                    pod_events.items, key=lambda x: x.last_timestamp
+                                ):
+                                    # Break out of loop if there are no new events
+                                    if (
+                                        event.last_timestamp
+                                        < self.job_pod_event_timestamps[job_name][
+                                            pod_name
+                                        ]
+                                    ):
+                                        break
+
+                                    self.job_pod_event_timestamps[job_name][
+                                        pod_name
+                                    ] = event.last_timestamp
+
+                                    pod_event_logs = [
+                                        f"Event: '{event.reason}' on pod '{pod_name}'"
+                                    ]
+                                    pod_event_logs.append(f"\tMessage: {event.message}")
```

Comment:
No need to build a list and join below, could build it all at once:

```suggestion
                                    log_msg = f"Event: {event.reason!r} on pod {pod_name!r}\n\tMessage: {event.message}"
```